src/main/java/org/logstash/dissect/fields/FieldFactory.java

```diff
@@ -0,0 +1,37 @@
+package org.logstash.dissect.fields;
+
+public final class FieldFactory {
+    private static final String MIXED_PREFIXES = "Field cannot prefix with both Append and Indirect Prefix (%s): %s";
+    private static final String PREFIXED_EMPTY = "Field cannot be a prefix on its own without further text";
+
+    public static Field create(String field) {
+        if (field.isEmpty() || field.startsWith("?")) {
```

Comment:
It is confusing.
The config as declared and loaded by LogStash is validated.

However the validators available are quite primitive and in this plugin's case only verifies that a Hash was declared but not whether the keys and values are valid.
The declared hash is given to this plugin at startup (register). It is here that I should really perform further validation but given that the Dissector class is created at this time, any format errors thrown by the Java constructor serve as validation.

So, no, there is no code based grammar processing on the Ruby side.

There is only documentation in Ruby comments in asciidoc form.